[Jodi Choo] iP

jodichoo opened the pull request:
### Duke

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is, 

1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features: 
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---------------
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

jodichoo committed:
add greet, echo, exit functions

jodichoo committed:
fix issue where sentence is treated as >1 input

jodichoo committed:
added TaskList class

jodichoo committed:
added 'list' function

jodichoo committed:
Change code logic of duke

jodichoo committed:
added a Task class

jodichoo committed:
accomodate new Task class + done() function to mark task

jodichoo committed:
Implement 'done' functionality

jodichoo committed:
add classes for the 3 types of tasks

jodichoo committed:
implement functionality to add 3 task types

jodichoo committed:
Wrote test

jodichoo committed:
Refactor some code in Duke

jodichoo committed:
Fulfill error handling minimum requirements

jodichoo committed:
handle errors when marking task as done

jodichoo committed:
Updated + ran tests

jodichoo committed:
represent task list as ArrayList instead of array

jodichoo committed:
added delete functionality

jodichoo committed:
Fix whitespace issues

jodichoo committed:
Updated test + minor code edits

jodichoo committed:
Fixed whitespace issues that interfere with tests

jodichoo committed:
Ignore tasks.txt save file

jodichoo committed:
Add methods to convert task to data for save file

jodichoo committed:
Add method to convert list to data for save file

jodichoo committed:
Change task-to-data conversion format

jodichoo committed:
Add constructor that takes in a ArrayList

jodichoo committed:
Add FileManager class, implement read/write from file

jodichoo committed:
Write to file when task is edited

jodichoo committed:
Store deadline date as LocalDate

jodichoo committed:
Handle exception when deadline input is invalid

jodichoo committed:
Merge branch 'branch-Level-7'

jodichoo committed:
Merge branch 'branch-Level-8'

jodichoo committed:
Add new Ui class

jodichoo committed:
Add new Parser class

jodichoo committed:
Refactor code to accomodate new classes

jodichoo committed:
Organize classes into duke package

jodichoo committed:
Update test file with new path to code

jodichoo committed:
Add JUnit tests for Event, Deadline, Todo, Parser classes

jodichoo committed:
Create jar file

jodichoo committed:
Add JavaDoc comments for public methods

jodichoo committed:
Add method to check for keyword in task description

jodichoo committed:
Add method that returns tasks containing keyword

jodichoo committed:
Add method to display tasks with keyword match

jodichoo committed:
Edit Duke and Parser class to recognise the 'find' command

jodichoo committed:
Add line break as the line was too long

jodichoo committed:
Merge branch 'branch-A-JavaDoc'

jodichoo committed:
Merge branch 'branch-A-CodingStandard'

jodichoo committed:
Merge branch 'branch-Level-9'

jodichoo committed:
Merge remote-tracking branch 'origin/add-gradle-support'

jodichoo committed:
Add Gradle support

jodichoo committed:
Add checkstyle config files

jodichoo committed:
Fix coding style violations

jodichoo committed:
Fix coding standard violations for tests

jodichoo committed:
Include JavaFX dependencies

Shruthi0310 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+     *
+     * @return The formatted data as a string.
+     */
+    public String toData() {
```
Perhaps could name it like formatData() to explain the method using the name

jodichoo replied:
Agree that it would be clearer, will take note of this!

Shruthi0310 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,111 @@
+        return command.equals("todo") || command.equals("deadline") || command.equals("event");
+    }
+
+    private String[] getAddTaskArgs() throws DukeException {
```
Maybe you could use switch statements instead of a lot of if else in order to make the code more readable

Shruthi0310 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,111 @@
+        return command.equals("todo") || command.equals("deadline") || command.equals("event");
+    }
+
+    private String[] getAddTaskArgs() throws DukeException {
```
Perhaps you could try to make the code shorter because currently it's more than 30 LOC

jodichoo replied:
I agree, have refactored some code to keep it shorter.

Shruthi0310 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,77 @@
+     * for them.
+     *
+     * @param list The list of matching tasks to be displayed.
+     */
+    public void displayMatching(TaskList list) {
```
Perhaps could be more specific with the name eg. displayMatchingTask

Shruthi0310 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,111 @@
+    }
+
+    private boolean isAddingNewTask() {
+        return command.equals("todo") || command.equals("deadline") || command.equals("event");
```
Perhaps this should be done is steps. For example
```
boolean isTodo = command.equals("todo");
boolean isDeadline = command.equals("deadline");
boolean isEvent = command.equals("event");
return isTodo || isDeadline || isEvent;
```

jodichoo replied:
Thanks for the suggestion, will make the change 👍

Shruthi0310 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,111 @@
+     * @return An array of Strings with valid commands and arguments.
+     * @throws DukeException If an invalid command or argument is given as user input.
+     */
+    public String[] parse() throws DukeException {
```
Same issue as mentioned earlier that too many if else might make it harder to read

jodichoo replied:
Okay, will take note

Shruthi0310 commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,55 @@
+
+import org.junit.jupiter.api.Test;
+
+public class ParserTest {
```
I like how the methods are written neatly and is very easy to read and understand!

Shruthi0310 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,75 @@
+    /**
+     * A method that runs the program.
+     */
+    public void run() {
```
I like the good use of SLAP to ensure that the code is at the same level of abstraction

Shruthi0310 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,62 @@
+     * @param keyword The keyword to be searched for in the description.
+     * @return Boolean indicating whether the keyword is found in the description.
+     */
+    public boolean contains(String keyword) {
```
Should this be something like hasKeyword since it is a boolean?

jodichoo replied:
Must have missed this, thanks for pointing it out!

Shruthi0310 reviewed (commented):
Hey! Overall, your code was generally easy to read and good coding quality except for some parts where there were too many if else statements. There weren't major violations except some minor ones that I have pointed out.😃

jodichoo committed:
Add classes to represent different commands

jodichoo committed:
Refactor code into new command classes

jodichoo committed:
Add Duke and User images for GUI

jodichoo committed:
Modify main class specified in build file

jodichoo committed:
Add GUI elements to Duke

jodichoo committed:
Merge branch 'branch-Level-10'

Shivlock221b commented on src/main/java/DialogBox.java:
```diff
@@ -0,0 +1,59 @@
+import java.util.Collections;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
```
I think these should be coupled in groups of 2.
Like
import *****
import *****

import *****
import *****

Just check once

Shivlock221b commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,22 @@
+public class DukeException extends Exception {
+    /**
+     * Constructor for a DukeException.
+     *
```
I think there should be a space between class definition and the function header

Shivlock221b commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,77 @@
+     * A method to stop user input.
+     */
+    public void stopInput() {
+        System.out.println("Byebye");
```
I think "stopInput" may not be correct name. Maybe "closeBot" or "systemExit". Just a suggestion

Shivlock221b reviewed (commented):
The code is very neatly presented and naming is done accurately, apart from some inaccuracies. I would like to replicate such a style in my code.

jodichoo committed:
Delete Ui class

jodichoo committed:
Modify gradle build file to enable assertions

jodichoo committed:
Add assertions to relevant points in code

jodichoo committed:
Fix mistake in assertion statement

jodichoo committed:
Fix missing and incorrect Javadocs

jodichoo committed:
getListFromFile method: extract a method getTask

getListFromFile method is too long (>30 LOC).

Long methods impact code quality.

Let's extract statements which work to instantiate a new Task and put them into
a new method named getTask.

jodichoo committed:
Remove and replace redundant print statements

jodichoo committed:
Merge pull request #1 from jodichoo/branch-A-Assertions

Use assertions

jodichoo committed:
Merge branch 'master' into branch-A-CodeQuality

jodichoo committed:
Merge pull request #2 from jodichoo/branch-A-CodeQuality

Improve code quality

jodichoo committed:
Remove text-based tests

jodichoo committed:
Set up continuous integration using GitHub Actions

jodichoo committed:
Fix checkstyle violations

jodichoo committed:
Implement sort command that sorts the list by task.

Tasks in the task list are displayed in the order they are added.

This may make it difficult for users to distinguish different task types.

Implementing a sort function allows the user to easily distinguish between tasks
of different types, making the application more user-friendly.

Let's
* Add a method in each task class to get their respective task icon.
* Add a method in TaskList class to sort the task list according to task
  character values.
* Create a new SortCommand class to represent the user command to sort the list.
* Edit parse method in Parser class to recognise the new sort command.

jodichoo committed:
Fix tests in ParserTest class

jodichoo committed:
Fix checkstyle violations

jodichoo committed:
Fix bug in computing the length of user input

jodichoo committed:
Remove I/O redirection tests from workflows

jodichoo committed:
Fix checkstyle violations

jodichoo committed:
Replace user profile picture

jodichoo committed:
Fix truncated dialog issue

Text will get cut off and replaced with ellipsis if the response is long.

Users are unable to properly view Duke's responses.

Replace Label object with Text object to contain the dialog.

jodichoo committed:
Tweak spacing and alignment for GUI

jodichoo committed:
Edit style of dialog boxes

jodichoo committed:
Handle cases where user input is empty or whitespace

jodichoo committed:
Implement pink dialog boxes for error messages

jodichoo committed:
Fix bug in determining error messages

jodichoo committed:
Reduce size of profile pictures

jodichoo committed:
Edit JavaDoc comments

jodichoo committed:
Fix checkstyle violations

jodichoo committed:
Fix bug in DeadlineCommand and EventCommand

Mistake in getDeadline and getEvent methods.

Missing deadline or time are determined incorrectly, resulting in
ArrayOutOfBounds error.

Fixed the mistake by editing the conditional statements in the
affected methods.

jodichoo committed:
Updated JUnit tests

jodichoo committed:
Set product name in title bar of window

jodichoo committed:
Add product screenshot

jodichoo committed:
Update README.md

jodichoo committed:
Fix emoji header bug

jodichoo committed:
Remove unnecessary print statements
